x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/ChatCompletionResponseEntity.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.response;
+
+import org.elasticsearch.inference.InferenceServiceResults;
+import org.elasticsearch.xpack.core.inference.results.ChatCompletionResults;
+import org.elasticsearch.xpack.inference.external.http.HttpResult;
+import org.elasticsearch.xpack.inference.external.http.retry.ResponseParser;
+import org.elasticsearch.xpack.inference.external.request.Request;
+
+import java.io.IOException;
+
+/**
+ * A base class for applying a subclassed "fromResponse" method to process results
+ * from a chat completion process. This is a start to abstract away from direct static methods
+ * for the "fromResponse" method in the apply function as used in other inference and provider types.
+ */
+public abstract class ChatCompletionResponseEntity implements ResponseParser {
```

Comment:
hmm, since this class and `EmbeddingResponseEntity` are pretty similar for now, how do you feel about collapsing them into a single class?

I think we can achieve that by doing something like:

```
protected abstract InferenceServiceResults fromResponse(Request request, HttpResult response) throws IOException;

...
```

Or I suppose we could omit this class and have the child classes implement `ResponseParser` directly?

My reasoning for using `InferenceServiceResults` is because parsers like cohere can return different types (`TextEmbeddingResults` and `TextEmbeddingByteResults`).

Reply:
Yeah - I think this makes sense to have a single base class for these two inference types (and more in the future) - I'll combine them. I thought about implementing ResponseParser directly, but I think that'd lead to more duplication of code at the end of the day, and think this method provides a nice, but maintainable wrapper...